ross/rotor_assembly.py

```diff
@@ -365,6 +367,44 @@ def flatten(l):
             + 2 * len([el for el in point_mass_elements])
         )
 
+        # global indexes for dofs
+        for elm in self.elements:
```

Comment:
I think we can improve this part of the code doing something like:
```python
n = 0
for elm in self.elements:
    # get the local_index tuple
    local_index = elm.dof_local_index()
    # get the global index by iterating on the local_index and adding n
    global_index = dof_tuple({k: v + n for k, v in local_index._asdict().items()})
    # set n as the last value from the global index
    n = global_index[-1]
```
Also, we should avoid checking types with `if elm.__class__.__name__ == '<type>'`. 
It is better to use `isinstance()` if necessary. For example, for the bearing case you would have `isinstance(elm, BearingElement)` and this would work for all types described in `bearing_class_list`, since they all inherit from `BearingElement`. This way, you would not need to update this list everytime a new bearing type is added.

But I think the type checking can all be avoided in this case, since we only need `n` from the last global index and the local index for the current element.

Reply:
I understood the isistance() part and I've modified it. But I didn't get the code suggestion right.
Should I replace all the code lines inside the `for` loop (line 370) by the code you wrote or just add these lines? I'm asking because `dof_tuple` would be referenced before assignment this way.
```
n = 0
for elm in self.elements:
    local_index = elm.dof_local_index()

    # in the line below dof_tuple is not defined
    global_index = dof_tuple({k: v + n for k, v in local_index._asdict().items()})

    n = global_index[-1]
```